src/fireedge/src/modules/components/Forms/Vm/CreateSchedActionForm/fields.js

```diff
@@ -0,0 +1,610 @@
+    const periodicValues = isVM
+      ? VM_SCHEDULE_TYPE_STRING
+      : TEMPLATE_SCHEDULE_TYPE_STRING
+
+    const periodicTypes = arrayToOptions(Object.keys(periodicValues), {
+      addEmpty: false,
+      getText: (key) => periodicValues[key],
+      getValue: (key) => key,
+    })
+
+    return periodicTypes
+  },
+  validation: string()
+    .trim()
+    .required()
+    .default(() => SCHEDULE_TYPE.ONETIME),
+  grid: { md: 12 },
+  notNull: true,
+})
+// --------------------------------------------------------
+// One time fields
+// --------------------------------------------------------
+
+/** @type {Field} Time field */
+const TIME_FIELD = {
+  name: 'TIME',
+  label: (typeAction) =>
+    typeAction === SCHEDULE_TYPE.PERIODIC ? T.StartTime : T.Time,
+  type: INPUT_TYPES.TIME,
+  dependOf: PERIODIC_FIELD_NAME,
+  htmlType: (typeAction) =>
+    typeAction === SCHEDULE_TYPE.RELATIVE && INPUT_TYPES.HIDDEN,
+  validation: date()
+    .min(getNow().toJSDate())
+    .transform(parseDateString)
+    .afterSubmit(dateToMilliseconds)
+    .default(() => getTomorrowAtMidnight())
+    .when(PERIODIC_FIELD_NAME, (typeAction, schema) =>
+      typeAction !== SCHEDULE_TYPE.RELATIVE ? schema.required() : schema
+    ),
+  fieldProps: {
+    minDateTime: getNow(),
+  },
+  grid: { md: 12 },
+}
+// --------------------------------------------------------
+// Periodic fields
+// --------------------------------------------------------
+
+/** @type {Field} Granularity of action */
+const REPEAT_FIELD = {
+  name: 'REPEAT',
+  label: T.GranularityOfAction,
+  type: INPUT_TYPES.AUTOCOMPLETE,
+  optionsOnly: true,
+  values: arrayToOptions(Object.keys(REPEAT_VALUES), {
+    addEmpty: true,
+    getText: (key) => sentenceCase(key),
+    getValue: (key) => REPEAT_VALUES[key],
+  }),
+  dependOf: PERIODIC_FIELD_NAME,
+  htmlType: (typeAction) =>
+    typeAction !== SCHEDULE_TYPE.PERIODIC && INPUT_TYPES.HIDDEN,
+  validation: string()
+    .trim()
+    .when(PERIODIC_FIELD_NAME, (typeAction, schema) =>
+      typeAction === TEMPLATE_SCHEDULE_TYPE_STRING.PERIODIC
+        ? schema.required()
+        : schema
+    ),
+  grid: { md: 12 },
+  notNull: true,
+}
+
+/** @type {Field} Weekly field */
+const WEEKLY_FIELD = {
+  name: 'WEEKLY',
+  dependOf: [PERIODIC_FIELD_NAME, REPEAT_FIELD.name],
+  type: INPUT_TYPES.AUTOCOMPLETE,
+  optionsOnly: true,
+  multiple: true,
+  label: T.DayOfWeek,
+  values: arrayToOptions(DAYS_OF_WEEK, {
+    addEmpty: false,
+    getValue: (_, index) => String(index),
+  }),
+  htmlType: (_, context) => {
+    const values = context?.getValues() || {}
+
+    return (
+      !(
+        values?.PERIODIC === SCHEDULE_TYPE.PERIODIC &&
+        values?.REPEAT === REPEAT_VALUES.WEEKLY
+      ) && INPUT_TYPES.HIDDEN
+    )
+  },
+  validation: lazy((_, { context }) =>
+    array(string())
+      .min(1)
+      .default(() => context?.[DAYS_FIELD.name]?.split?.(',') ?? [])
+      .when(REPEAT_FIELD.name, (repeatType, schema) =>
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This route is missing authorization checks. Without proper authorization, sensitive routes may be exposed to unauthorized users, leading to potential data breaches or unauthorized actions. Ensure that authorization middleware or handlers are applied. For more information on missing authorization vulnerabilities, refer to [CWE-862](https://cwe.mitre.org/data/definitions/862.html).